### PR TITLE
Add dependency injection [001]

### PR DESCRIPTION
**Pull request by Lezzess**

### CODE REVIEW

**Comment by Lezzess** [comment]
[Add dependency injection [001]](https://trello.com/c/i5QqfzWC/3-add-dependency-injection-001)

File: src/Database/DependencyInjection/Dependencies.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Database.DependencyInjection
+{
+    public class Dependencies : NinjectModule
+    {
+        #region Public Methods
+
+        public override void Load()
+        {
+
```
Comment (ATL-95) [inline src/Database/DependencyInjection/Dependencies.cs:11]: We could use throw new NotImplementedException() until methods not implemented, i think it is a good behavior

File: src/UserInterface/UserInterface.Android/MainActivity.cs [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
+using UserInterface.DependencyInjection;
 
 namespace UserInterface.Droid
 {
     [Activity(Label = "UserInterface", Icon = "@mipmap/icon", Theme = "@style/MainTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation | ConfigChanges.UiMode | ConfigChanges.ScreenLayout | ConfigChanges.SmallestScreenSize )]
     public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
     {
+        #region Public Methods
+
+        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
```
Comment (ATL-95) [inline src/UserInterface/UserInterface.Android/MainActivity.cs:15]: If Permission class do not conflict with another class in another library, please add reference to it in top.

File: src/UserInterface/UserInterface.Android/UserInterface.Android.csproj [hunk 3]
```diff
@@ -89,10 +89,19 @@
...
       <Project>{60240989-4689-4202-A7FE-F4E080BBE9F3}</Project>
       <Name>UserInterface</Name>
     </ProjectReference>
   </ItemGroup>
- <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
-</Project>
+  <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
+  <ProjectExtensions>
+    <VisualStudio>
+      <UserProperties TriggeredFromHotReload="False" />
```
Comment (ATL-95) [inline src/UserInterface/UserInterface.Android/UserInterface.Android.csproj:104]: We will need be on this function for active development

**Review by ATL-95 — CHANGES_REQUESTED** [review]
You need look my comments and if you have different opinion we will need discussed it.

File: src/Database/DependencyInjection/Dependencies.cs (hunk 1 above)
Comment (Lezzess) [reply src/Database/DependencyInjection/Dependencies.cs:11]: Exception will crash the application

File: src/UserInterface/UserInterface.Android/UserInterface.Android.csproj (hunk 3 above)
Comment (Lezzess) [reply src/UserInterface/UserInterface.Android/UserInterface.Android.csproj:104]: I don't know what this property is for but hot reload works for me even with it being False.
It was autogenerated by Visual Studio so I did not change anything

**Review by ATL-95 — APPROVED** [review]

**Pull request merged**